### DISCUSSION TITLE
Is it possible to publish a jupyter book on gitlab pages?

### DISCUSSION THREAD

**Discussion by AndreGismo** [executablebooks/meta#471] [Q&A] [answered]
Hello community,

I was wondering, if it is possible to publish a jupyter book on gitlab pages.

I have allready published a jupyter book here in github pages, following the instructions [here](https://jupyterbook.org/start/publish.html#publish-your-book-online-with-github-pages), and all worked well - `ghp-import` takes care of everything.

Now I have also tried to publish a jupyter book via gitlab pages - but I'm confused, because this is a completely different approach, using a `.gitlab-ci.yml` file and a static site generator like jekyll (as described in [here](https://docs.gitlab.com/ee/user/project/pages/getting_started/pages_from_scratch.html)). I think I don't even need this jekyll, because jupyter book has already built the html.

And I have noticed that I get a warning when I run `jb build` and my `_config.yml` contains the following:

```
repository:
    url: https://gitlab.com/<user>/<repo>
    branch: main

```

The warning says:

```
WARNING: Currently Binder/JupyterHub repositories must be on GitHub
```

So I am wondering, if it is even possible (at least at the moment) to publish jupyter books on gitlab pages.

Has anyone experience with this?
Thanks for any for any help in advance!

**Comment by choldgraf** [comment, marked as answer]
yes! check out here for a working example: https://github.com/executablebooks/jupyter-book/issues/1416

for launch buttons that one will require future development I believe, but certainly is welcome

**Reply by AndreGismo** [reply]
Thank you for pointing me to this example @choldgraf !
Perfect, a small exaple, exactly what I was looking for.
Thanks! :)
